src/onegov/org/views/resource.py

```diff
@@ -93,13 +93,31 @@ def get_resource_form(self, request, type=None):
 @OrgApp.html(model=ResourceCollection, template='resources.pt',
              permission=Public)
 def view_resources(self, request, layout=None):
+    default_group = request.translate(_("General"))
     resources = group_by_column(
         request=request,
         query=self.query(),
+        default_group=default_group,
         group_column=Resource.group,
         sort_column=Resource.title
     )
 
+    def contains_at_least_one_room(resources):
+        for resource in resources:
+            if resource.type == 'room':
+                return True
+        return False
+
+    for group, entries in resources.items():
+        # don't include find-your-spot link for categories with
+        # no rooms
+        if not contains_at_least_one_room(entries):
+            continue
+        entries.insert(0, FindYourSpotCollection(
+            request.app.libres_context,
+            group=None if group == default_group else group
+        ))
```

Comment:
This should happen after `combine_grouped` otherwise "Find Your Spot" will be sorted alphabetically into the group, it should always stick to the top.

Also `combine_grouped` could be slightly optimized by using `itertools.chain` instead of the `+` operator so we don't create a temporary list before sorting.